lightning/src/ln/peers/handler.rs

```diff
@@ -0,0 +1,3280 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! Top level peer message handling and socket handling logic lives here.
+//!
+//! Instead of actually servicing sockets ourselves we require that you implement the
+//! SocketDescriptor interface and use that to receive actions which you should perform on the
+//! socket, and call into PeerManager with bytes read from the socket. The PeerManager will then
+//! call into the provided message handlers (probably a ChannelManager and NetGraphmsgHandler) with messages
+//! they should handle, and encoding/sending response messages.
+
+use bitcoin::secp256k1::key::{SecretKey,PublicKey};
+
+use ln::features::InitFeatures;
+use ln::msgs;
+use ln::msgs::{ChannelMessageHandler, LightningError, RoutingMessageHandler};
+use ln::channelmanager::{SimpleArcChannelManager, SimpleRefChannelManager};
+use util::ser::{Writeable};
+use ln::wire;
+use ln::wire::{Encode, Message};
+use util::byte_utils;
+use util::events::{MessageSendEvent, MessageSendEventsProvider};
+use util::logger::Logger;
+use routing::network_graph::NetGraphMsgHandler;
+
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::{cmp,error,hash,fmt};
+use std::ops::Deref;
+
+use bitcoin::hashes::sha256::Hash as Sha256;
+use bitcoin::hashes::sha256::HashEngine as Sha256Engine;
+use bitcoin::hashes::{HashEngine, Hash};
+use ln::peers::outbound_queue::OutboundQueue;
+use ln::peers::transport::{PayloadQueuer, Transport};
+use bitcoin::hashes::core::iter::Filter;
```

Comment:
This should be `std::iter::Filter`.

Reply:
Done: squashed into `refactor: Clean up the PeerHolder access/iteration/remove`